--- /home/duybui/.config/fisherman/bobthefish/fish_prompt.fish	2017-12-03 13:44:39.935629000 -0500
+++ /home/duybui/.config/fisherman/bobthefish/fish_prompt.fish	2017-12-03 13:44:31.302309800 -0500
@@ -263,6 +263,20 @@
       set directory '/'
     case "$HOME"
       set directory '~'
+    case "$HOME/code/*/src/*"
+      set parent    (string split "/" (string sub -s (math 7 + (string length $HOME)) "$current_dir"))[1]' '$right_arrow_glyph' '(pwe)
+      set directory (string sub -s (math 13 + (string length $HOME) + 4 + (string length $parent) - 2) "$current_dir")
+    case "$HOME/code/*"
+      set parent    (string split "/" (string sub -s (math 7 + (string length $HOME)) "$current_dir"))[1]
+      set directory (string sub -s (math 13 + (string length $HOME) + (string length $parent) + 1) "$current_dir")
+      [ (string length $directory) != 0 ]; and set parent $parent' '$right_arrow_glyph' '
+    case "/mnt/c/Users/$WUSER/code/*/src/*"
+      set parent    (string split "/" (string sub -s (math 20 + (string length $WUSER)) "$current_dir"))[1]' '$right_arrow_glyph' '(pwe)
+      set directory (string sub -s (math 20 + (string length $WUSER) + 4 + (string length $parent) - 2) "$current_dir")
+    case "/mnt/c/Users/$WUSER/code/*"
+      set parent    (string split "/" (string sub -s (math 20 + (string length $WUSER)) "$current_dir"))[1]
+      set directory (string sub -s (math 20 + (string length $WUSER) + (string length $parent) + 1) "$current_dir")
+      [ (string length $directory) != 0 ]; and set parent $parent' '$right_arrow_glyph' '
     case '*'
       set parent    (__bobthefish_pretty_parent "$current_dir")
       set directory (basename "$current_dir")
@@ -1395,5 +1400,22 @@
     __bobthefish_prompt_dir
   end

+  if test (which brazil 2>/dev/null)
+    __bobthefish_start_segment brmagenta white
+    echo -n "aps "
+
+    if klist -s
+    else
+      __bobthefish_start_segment red white
+      echo -n "!k "
+    end
+
+    if test (find ~/.midway -type f -ctime -0.5)
+    else
+      __bobthefish_start_segment brred white
+      echo -n "!mw "
+    end
+  end
+
   __bobthefish_finish_segments
 end
